### ISSUE TITLE
[Task] RSX tiled memory - remaining work

### ISSUE THREAD

**Issue by kd-11** [RPCS3/rpcs3#14682]
- [ ] Fix leaking paths in FrostBite2 titles e.g NFS Rivals
- [ ] OpenGL support

**Comment by legend800** [comment]
As someone mentioned in the original PR, Split Second doesn't render explosions/smoke (big part of the game). I just found out that they render ONLY if you use a higher res. But if you do that, then you have mega pixelation issues like this:

<img width="961" alt="2023-09-27 09_25_51-FPS_ 29 99 _ Vulkan _ 0 0 29-15655 _ Split_Second  NPUB30450" src="https://github.com/RPCS3/rpcs3/assets/20983844/aa000b91-1537-4411-9782-2a2304e6a369">

So native: missing effects, higher res=effects render but large pixels (similar to when tiled memory is off)
Settings: 
- Strict (fix red lines from AF + pixelated gfx form non-native res)
- WCB + RCB + RSX tiled memory (skybox smearing and env lighting)

**Comment by Ordinary205** [comment]
The Need for Speed Most Wanted/Hot Pursuit car reflections is fixed by enabling WCB+RSX tiled memory, but the reflections appears inverted.
Real PS3:
![NFS Hot Pursuit PS3](https://github.com/RPCS3/rpcs3/assets/125112932/1c6f37a4-c240-4ff9-b12a-2335ed156243)

RPCS3:
![NFS Hot Pursuit (2)](https://github.com/RPCS3/rpcs3/assets/125112932/239ca612-87af-4001-980e-50f34a1e8827)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/12756449/RPCS3.log.gz)

**Comment by kd-11** [comment]
I reproduced the split second issue. It is likely a codepath issue, since using any scaling including 99% or 101% makes the smoke appear again. I'll debug the real cause later in the week.

**Comment by Linear524** [comment]
Some bugs to solve for fully working MLAA in Uncharted 3 and TLoU :
![1001](https://github.com/RPCS3/rpcs3/assets/84367301/fd3a934c-63bf-4b16-a59c-ed3c440d85cd)
![1002](https://github.com/RPCS3/rpcs3/assets/84367301/f47128ec-9d01-46af-9249-64b5ad6db888)

Those buggy horizontal lines consist of garbled mask pixels from overall scene - if you move camera a little during those scenes - pixel objects inside those horizontal lines will move as well...
Looks like a bit wrong memory sections to read or write MLAA data.

**Comment by kd-11** [comment]
The stripes should be fixed in the next PR. Actually I probably should just submit that work. I was hoping to knock down a few regressions at once but no need making everything too complicated.

**Comment by kd-11** [comment]
Test with https://github.com/RPCS3/rpcs3/pull/14706

**Comment by Linear524** [comment]
@KD-11
TLoU and Uncharted 3 stripes still not fixed with #14706

**Comment by kd-11** [comment]
Any errors in the log?

**Comment by Linear524** [comment]
Will try to collect logs today :)

**Comment by Linear524** [comment]
@KD-11
Nope - nothing in the logs, just standard PPU trampoline spam messages...

**Comment by Ordinary205** [comment]
Enabling Handle RSX tiling memory partially fixes the other lighting for Army of TWO: The Devil's Cartel, but this causes more rainbow graphics to appear on the left side.
WCB+RCB:
![Army of TWO WCB+RCB](https://github.com/RPCS3/rpcs3/assets/125112932/1a7feed1-158b-452c-ba18-942f2cadf73f)

WCB+RCB+RSX tiling memory (disabling the MLAA is required to remove yellow graphics):
![Army of TWO WCB+RCB+RSX tiling memory](https://github.com/RPCS3/rpcs3/assets/125112932/01837f46-53b1-4e10-9eb5-a6a4eae7fab6)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/12839934/RPCS3.log.gz)
Testing the PR named "rsx: Detiler improvements" doesn't show any differences.

**Comment by Ordinary205** [comment]
> This PR no longer fixes NFS The Run, and Battlefield 3 now looks like this.
![Battlefield 3](https://github.com/RPCS3/rpcs3/assets/125112932/831fd662-4f1a-4fdb-ad5c-94c6735e9cc1)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/12882795/RPCS3.log.gz)


I think I accidently tested the wrong build which was 15683 instead of 15684, but I'm unable to confirm if this improves regression on frostbite games because the windows build is unavailable.

**Comment by Ordinary205** [comment]
Thanks to illusion for messaging me on discord to get the latest build, I can now confirm that NFS The Run and Battlefield 3 hasn't reverted.
![15684](https://github.com/RPCS3/rpcs3/assets/125112932/8adb23e7-7def-422a-b5b0-33383a06ee70)
![15684 battlefield 3](https://github.com/RPCS3/rpcs3/assets/125112932/0c4b5e94-3fd3-4dba-82db-db33ce11e25f)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/12884840/RPCS3.log.gz)

**Comment by franklypaladin** [comment]
Fixes Gran Turismo 5's crowd being blue blobs until you are right next to them issue.

**Comment by Ordinary205** [comment]
Enabling RSX tiled memory causes NFS Most Wanted to have more square artifacts to appear on top of the left/right side on the screen.
Before: (WCB)
![NFS Most Wanted](https://github.com/RPCS3/rpcs3/assets/125112932/8fbb4358-8220-4734-93a2-d66356e1838e)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/13000719/RPCS3.log.gz)

After: (WCB+RSX tiled memory)
![NFS Most Wanted RSX tiled memory](https://github.com/RPCS3/rpcs3/assets/125112932/8e93332d-c309-4018-b3fa-a038e7ebac04)
[RSX tiled memory.gz](https://github.com/RPCS3/rpcs3/files/13000791/RSX.tiled.memory.gz)
This issue doesn't happen on NFS hot pursuit.

**Comment by kd-11** [comment]
This one is a race condition between SPU post-fx pipe and RSX post-fx pipe. The only solution is to speed up SPU code drastically while WCB is active which isn't really feasible without some low-level changes. Something can definitely be done for AMD cards which support device-coherent memory, but for the rest, only faster CPUs will save us.
You can also give "Accurate RSX reservations" toggle a try and see if it helps stabilize the issue.
This problem already exists in other games such as GT6 as well.

**Comment by Ordinary205** [comment]
Enabling RSX tiled memory causes a regression on NFS Hot Pursuit when having read color buffers and asynchronous texture streaming enabled.
Before: WCB+RCB+ATS
![NFS Hot Pursuit before](https://github.com/RPCS3/rpcs3/assets/125112932/ea6a1925-3128-4a34-ad4e-3e6d499fddcf)

After: WCB+RCB+ATS+RSX tiled memory
![NFS Hot Pursuit after](https://github.com/RPCS3/rpcs3/assets/125112932/403a3db4-9a3d-483b-a39e-5c400891ade0)

But if I disable the read color buffers or asynchronous texture streaming, the regression is gone.
![NFS Hot Pursuit without](https://github.com/RPCS3/rpcs3/assets/125112932/226ae203-7d97-42c5-a62a-9083160a8ec5)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/13052538/RPCS3.log.gz)

**Comment by kd-11** [comment]
CPU blit is now feature complete as of https://github.com/RPCS3/rpcs3/pull/14934
There are a lot of corner cases when GPU blit is active, I'll try and get those sorted in coming weeks but the limitation is well known (surface and texture cache are separate and don't communicate).
I think a lot of corner cases can be handled by replacing the RCB load with a texture cache load+copy (basically a fullscreen quad draw with texture cache as the source). Doing that is easier said than done however.

**Comment by Ordinary205** [comment]
NFS Most Wanted has stability issues when enabling RCB or RSX Tiled memory before going ingame
If I enable RCB with WCB+RSX Tiled memory, the game crashes with this error:
WCB+RCB+RSX Tiled memory:
![Need for Speed Most Wanted doesnt](https://github.com/RPCS3/rpcs3/assets/125112932/b656335b-286e-4fc5-9289-2c67754ec8d0)
![Need for Speed Most Wanted error](https://github.com/RPCS3/rpcs3/assets/125112932/eba0d3e2-e8c0-407e-9b8c-202bef2789d7)

However if I try disabling RCB or RSX Tiled memory, the game works fine and enters ingame successfully.
WCB+RSX Tiled memory:
![Need for Speed Most wanted working](https://github.com/RPCS3/rpcs3/assets/125112932/b1ff3b64-13c7-4aab-a77b-b64622363339)

WCB+RCB+RSX Tiled memory:
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/13797496/RPCS3.log.gz)

**Comment by kd-11** [comment]
> NFS Most Wanted has stability issues when enabling RCB or RSX Tiled memory before going ingame If I enable RCB with WCB+RSX Tiled memory, the game crashes with this error: WCB+RCB+RSX Tiled memory: ![Need for Speed Most Wanted doesnt](https://private-user-images.githubusercontent.com/125112932/293450311-b656335b-286e-4fc5-9289-2c67754ec8d0.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQwNDQwNzIsIm5iZiI6MTcwNDA0Mzc3MiwicGF0aCI6Ii8xMjUxMTI5MzIvMjkzNDUwMzExLWI2NTYzMzViLTI4NmUtNGZjNS05Mjg5LTJjNjc3NTRlYzhkMC5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjMxMjMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIzMVQxNzI5MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xYmM1N2FkMzdiZWJhMjhkZTkzNzYyMTYzMzUxZDlmMDliMmIxNTMxNjU2NTE2YTM4N2VhNWRlNjAxZWJkNTNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.d78lR2N3jeNmJpqb07A6v-D7rTqFvIYlvt_UmUO6EGE) ![Need for Speed Most Wanted error](https://private-user-images.githubusercontent.com/125112932/293450315-eba0d3e2-e8c0-407e-9b8c-202bef2789d7.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQwNDQwNzIsIm5iZiI6MTcwNDA0Mzc3MiwicGF0aCI6Ii8xMjUxMTI5MzIvMjkzNDUwMzE1LWViYTBkM2UyLWU4YzAtNDA3ZS05YjhjLTIwMmJlZjI3ODlkNy5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjMxMjMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIzMVQxNzI5MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jOTM0MmU1Mjk3NGM5OWMyNGM0ODA1MTM5YzMwOGJlMDY2NDcwYzBhMmViYmU1MzA4ZmZkMGUzOGExZDczODQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Z2HCthdREkj9_bO5DDmqaOMfll78okEFZOnNQj_bFiY)
> 
> However if I try disabling RCB or RSX Tiled memory, the game works fine and enters ingame successfully. WCB+RSX Tiled memory: ![Need for Speed Most wanted working](https://private-user-images.githubusercontent.com/125112932/293450328-b1ff3b64-13c7-4aab-a77b-b64622363339.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQwNDQwNzIsIm5iZiI6MTcwNDA0Mzc3MiwicGF0aCI6Ii8xMjUxMTI5MzIvMjkzNDUwMzI4LWIxZmYzYjY0LTEzYzctNGFhYi1hNzdiLWI2NDYyMjM2MzMzOS5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjMxMjMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIzMVQxNzI5MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YTJmNjcwY2Y1OTgyM2M5MmE5MGI5OWFlMmU0ZTU4OTRjYzBiMjE5MzdhMzQ5Mjk1YzY0NWI2N2I3ODhjZjUzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1o1OZfedBLiiHW6cgqJVbZPG6dBgl9Qt6gTzUu6rFDY)
> 
> WCB+RCB+RSX Tiled memory: [RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/13797496/RPCS3.log.gz)

This should be a separate ticket.

**Comment by Ordinary205** [comment]
> > NFS Most Wanted has stability issues when enabling RCB or RSX Tiled memory before going ingame If I enable RCB with WCB+RSX Tiled memory, the game crashes with this error: WCB+RCB+RSX Tiled memory: ![Need for Speed Most Wanted doesnt](https://private-user-images.githubusercontent.com/125112932/293450311-b656335b-286e-4fc5-9289-2c67754ec8d0.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQwNDQwNzIsIm5iZiI6MTcwNDA0Mzc3MiwicGF0aCI6Ii8xMjUxMTI5MzIvMjkzNDUwMzExLWI2NTYzMzViLTI4NmUtNGZjNS05Mjg5LTJjNjc3NTRlYzhkMC5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjMxMjMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIzMVQxNzI5MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xYmM1N2FkMzdiZWJhMjhkZTkzNzYyMTYzMzUxZDlmMDliMmIxNTMxNjU2NTE2YTM4N2VhNWRlNjAxZWJkNTNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.d78lR2N3jeNmJpqb07A6v-D7rTqFvIYlvt_UmUO6EGE) ![Need for Speed Most Wanted error](https://private-user-images.githubusercontent.com/125112932/293450315-eba0d3e2-e8c0-407e-9b8c-202bef2789d7.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQwNDQwNzIsIm5iZiI6MTcwNDA0Mzc3MiwicGF0aCI6Ii8xMjUxMTI5MzIvMjkzNDUwMzE1LWViYTBkM2UyLWU4YzAtNDA3ZS05YjhjLTIwMmJlZjI3ODlkNy5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjMxMjMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIzMVQxNzI5MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jOTM0MmU1Mjk3NGM5OWMyNGM0ODA1MTM5YzMwOGJlMDY2NDcwYzBhMmViYmU1MzA4ZmZkMGUzOGExZDczODQyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Z2HCthdREkj9_bO5DDmqaOMfll78okEFZOnNQj_bFiY)
> > However if I try disabling RCB or RSX Tiled memory, the game works fine and enters ingame successfully. WCB+RSX Tiled memory: ![Need for Speed Most wanted working](https://private-user-images.githubusercontent.com/125112932/293450328-b1ff3b64-13c7-4aab-a77b-b64622363339.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQwNDQwNzIsIm5iZiI6MTcwNDA0Mzc3MiwicGF0aCI6Ii8xMjUxMTI5MzIvMjkzNDUwMzI4LWIxZmYzYjY0LTEzYzctNGFhYi1hNzdiLWI2NDYyMjM2MzMzOS5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjMxMjMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIzMVQxNzI5MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YTJmNjcwY2Y1OTgyM2M5MmE5MGI5OWFlMmU0ZTU4OTRjYzBiMjE5MzdhMzQ5Mjk1YzY0NWI2N2I3ODhjZjUzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1o1OZfedBLiiHW6cgqJVbZPG6dBgl9Qt6gTzUu6rFDY)
> > WCB+RCB+RSX Tiled memory: [RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/13797496/RPCS3.log.gz)
> 
> This should be a separate ticket.

Well I thought it would be a good idea to report this issue on this ticket, since its an RSX tiled memory bug, but I'll make a report anyways.

**Comment by kd-11** [comment]
I create the tickets for specific work items but these are usually theoretical (i.e no logs). This error is unexpected hence the assert rather than just broken graphics.

**Comment by kd-11** [comment]
With https://github.com/RPCS3/rpcs3/pull/14985, frostbite now renders semi-correctly. The only issue left to resolve is the texture tiling (repeating) which is a known blit-engine issue that I was never able to observe directly.
![image](https://github.com/RPCS3/rpcs3/assets/15904127/0439da5e-cf0a-48b1-b7c7-f771f547a16d)

**Comment by kd-11** [comment]
https://github.com/RPCS3/rpcs3/pull/14985 also fixes split/second issue with missing effects. That was being tracked by https://github.com/RPCS3/rpcs3/issues/14984

**Comment by legend800** [comment]
Gfx broken on Split Second with this PR (100%):

<img width="961" alt="2023-12-31 15_05_13-FPS_ 23 31 _ Vulkan _ 0 0 29-705e17ae Alpha _ maint0 _ Split_Second  NPUB30450" src="https://github.com/RPCS3/rpcs3/assets/20983844/750011be-cea4-4393-a414-78b6e8dcce66">
<img width="960" alt="2023-12-31 15_12_46-FPS_ 29 99 _ Vulkan _ 0 0 29-705e17ae Alpha _ maint0 _ Split_Second  NPUB30450" src="https://github.com/RPCS3/rpcs3/assets/20983844/b12782d7-b3a0-432c-8007-3c8b7a538695">

Settings: Strict(fix red lines), WCB + RCB + RSX tiled memory(skybox smearing and env lighting), 100% res(pixelation)

[Split Second [NPUB30450].log.gz](https://github.com/RPCS3/rpcs3/files/13802347/Split.Second.NPUB30450.log.gz)

**Comment by kd-11** [comment]
That was always inconsistent for me even before the PR. Likely some timing thing.
In my case I use all the buffer options and tiled mem option. It could be related to the mirrors effect bug but that's something else.

**Comment by legend800** [comment]
R/W depth buffers aren't needed. Results are 100% consistent, so I don't think there's a race condition here. Gfx broken on PR and fine on the latest Master build (aside from what's noted here https://github.com/RPCS3/rpcs3/issues/14984):

Master:
<img width="963" alt="2024-01-01 09_15_17-" src="https://github.com/RPCS3/rpcs3/assets/20983844/f6ef6a48-4104-44ef-a580-71eb96ce9b6b">
<img width="963" alt="2024-01-01 09_14_36-FPS_ 30 01 _ Vulkan _ 0 0 29-15898 _ Split_Second  NPUB30450" src="https://github.com/RPCS3/rpcs3/assets/20983844/a2bed4ff-f51e-4df8-971f-29d4ecca5aea">

[Split Second [NPUB30450].log.gz](https://github.com/RPCS3/rpcs3/files/13805401/Split.Second.NPUB30450.log.gz)

**Comment by legend800** [comment]
@kd-11 Did you see this comment here: https://github.com/RPCS3/rpcs3/issues/14984#issuecomment-1873414256? 

Game was not improved and that PR only regressed it. This is not accurate:

![image](https://github.com/RPCS3/rpcs3/assets/20983844/3b8ee594-fcbb-4caf-9245-05c666d30632)

**Comment by Asinin3** [comment]
Why tag someone three times about the same thing.

**Comment by legend800** [comment]
Because it was ignored and I'm trying to get attention on it.

**Comment by Asinin3** [comment]
@legend800  Once is enough, don't spam.

**Comment by kd-11** [comment]
Vertical repeats and flickering at the bottom of the screen (star wars) are caused by tile encoding into partial 2D regions. This is not allowed because the spatial distribution is not coherent. A linear 2D region scatters the pixels all around in X and Y with the exception that Y regions seem to behave. Either way, some changes are needed to ensure the whole tile is available to the encoder:
[ ] Encoder block should overwrite the mapping and basically just render to tile.
[ ] DMA sections created from tiles must preload the entire tile. Memory operations make no sense otherwise.

With halfplemented variants of the above the mirror/repeat bug disappears from NFS:Rivals and Split/Second.

**Comment by kd-11** [comment]
I've implemented the requirements above in https://github.com/RPCS3/rpcs3/pull/15160
However, there is now a strange bug where partial writes cause flicking in the top and bottom parts of the frame.
Loading the entire tile from offset 0 hides the bug so I don't know what is actually causing it.

**Comment by kd-11** [comment]
I need to look into this one again before I move on to other things. It's a core feature for most AAAs when patches are not in use (required for official playable status).

**Comment by Darkhost1999** [comment]
Should this rsx memory tiling setting in theory be always enabled. While the feature isn't complete sure disable and keep in advanced tab but eventually leave enabled and in advanced tab then upon no reported issues migrate to debug tab then remove setting.
Yet the question is primarily will this hurt games that don't need this option and only help games that do need this option?

**Comment by kd-11** [comment]
Once the feature is completed it will be enabled always. I still plan on returning to this, just need to run down a few more regressions first.
